pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -71,7 +74,20 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 		return toAdmissionResponseError(err)
 	}
 
-	pvcSource := dataVolume.Spec.Source.PVC
+	if dataVolume.Spec.Source != nil {
+		pvcSource = dataVolume.Spec.Source.PVC
+	} else if dataVolume.Spec.SourceRef != nil && dataVolume.Spec.SourceRef.Kind == cdiv1.DataVolumeDataSource {
+		ns := dataVolume.Namespace
+		if dataVolume.Spec.SourceRef.Namespace != nil && *dataVolume.Spec.SourceRef.Namespace != "" {
+			ns = *dataVolume.Spec.SourceRef.Namespace
+		}
+		dataSource, err := wh.cdiClient.CdiV1beta1().DataSources(ns).Get(context.TODO(), dataVolume.Spec.SourceRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return toAdmissionResponseError(err)
+		}
+		pvcSource = dataSource.Spec.Source.PVC
```

Comment:
Isn't it possible, that a DataSource doesn't point to a PVC yet?
Will this cause any problems here?

Reply:
Yeah should make sure that's not nil

Reply:
WRT nil PVC, I don't see any difference from the original one a few lines above:
```
if dataVolume.Spec.Source != nil {
    pvcSource = dataVolume.Spec.Source.PVC
} ...
```
Anyway we have a check a few lines below:
```
if pvcSource == nil {
    klog.V(3).Infof("DataVolume %s/%s not cloning", targetNamespace, targetName)
    return allowedAdmissionResponse()
}
```